packages/lisk-transactions/src/base_transaction.ts

```diff
@@ -305,6 +307,65 @@ export abstract class BaseTransaction {
 		return createResponse(this.id, errors);
 	}
 
+	public addMultisignature(
+		store: StateStore,
+		signatureObject: SignatureObject,
+	): TransactionResponse {
+		// Get the account
+		const account = store.account.get(this.senderId);
+		// Check if signature is not already there
+		if (this.signatures.includes(signatureObject.signature)) {
+			return createResponse(this.id, [
+				new TransactionError(
+					`Signature '${
+						signatureObject.signature
+					}' already present in transaction.`,
+					this.id,
+				),
+			]);
+		}
+
+		// Validate signature key belongs to account's multisignature group
+		if (
+			account.membersPublicKeys &&
+			!account.membersPublicKeys.includes(signatureObject.publicKey)
+		) {
+			return createResponse(this.id, [
+				new TransactionError(
+					`Public Key '${
+						signatureObject.publicKey
+					}' is not a member for account '${account.address}'.`,
+					this.id,
+				),
+			]);
+		}
+
```

Comment:
maybe if it's not included in the signatue already, we can just added to the `signatures` and `return this.processMulltisignature`?

Reply:
I'm not sure I understand the issue; if the PK that was sent is not one of the registered with the accounts should be rejected.

Reply:
I don't think we should add the signature to the array, and return the `processMultisignature` function because `processMultisignature`  will set the status to invalid for the transaction if the new signature is invalid. Which is not what we want.